data/hlint.yaml

```diff
@@ -240,6 +240,8 @@
     - warn: {lhs: x `mappend` mempty, rhs: x, name: "Monoid law, right identity"}
     - warn: {lhs: foldr (<>) mempty, rhs: mconcat}
     - warn: {lhs: foldr mappend mempty, rhs: mconcat}
+    - warn: {lhs: mempty x, rhs: mempty, name: Evaluate}
+    - warn: {lhs: mempty x y, rhs: mempty, name: Evaluate}
```

Comment:
Doesn't repeated application of `mempty x` catch this, by evaluating twice?

Reply:
If you mean to only catch ``x `mempty` y`` then I'd write that explicitly - and perhaps also give it the name "possible typo of mappend"

Reply:
I might even suggest ``x `mappend` y`` and give the Note "changes semantics likely because of a bug"

Reply:
> Doesn't repeated application of `mempty x` catch this, by evaluating twice?

Not unless/until #1096 is fixed.

> If you mean to only catch `` x `mempty` y `` then I'd write that explicitly - and perhaps also give it the name "possible typo of mappend"

Done.

> I might even suggest `` x `mappend` y `` and give the Note "changes semantics likely because of a bug"

I don't really want to do that for the same reason I mentioned in #1092.

Reply:
Could always do a `Note: Perhaps you meant to use mappend` even if we make the suggestion to mempty?

Reply:
Ah, I see you beat me to it!